### PR TITLE
feat: add PR and issue auto-labeling for triaging

### PR DESCRIPTION
**Pull request by torresxb1**
*Issue #, if available:*

*Description of changes:*
Copied over from the [SAM repo](https://github.com/aws/serverless-application-model), and created the relevant labels in this repo.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/pr-labeler.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  apply-internal-external-label:
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v5
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const maintainers = ['jfuss', 'c2tarun', 'hoffa', 'awood45', 'CoshUS', 'aahung', 'hawflau', 'mndeveci', 'ssenchenko', 'wchengru', 'mingkun2020', 'qingchm', 'moelasmar', 'xazhao', 'mildaniel', 'marekaiv', 'torresxb1']
```
Comment (qingchm) [inline .github/workflows/pr-labeler.yml:16]: Can we update the maintainers list here?

**Review by qingchm — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-labeler.yml (hunk 1 above)
Comment (torresxb1) [reply .github/workflows/pr-labeler.yml:16]: I'd prefer having follow-up PRs to update the maintainers list in all the repos

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+name: Feature request
+about: Suggest an idea/feature/enhancement for AWS SAM Build Images
```
Comment (hoffa) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
about: Suggest an idea/feature/enhancement for AWS SAM build images
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+name: Bug report
+about: Create an issue to report a bug for AWS SAM Build Images
```
Comment (hoffa) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
about: Create an issue to report a bug for AWS SAM build images
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+---
+
+<!-- Make sure we don't have an existing Issue for that feature request (open or closed). -->
+
+### Describe your idea/feature/enhancement
+
+Provide a clear description.
+
+Ex: I wish AWS SAM Build Images would [...]
```
Comment (hoffa) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
e.g., I wish AWS SAM build images would [...]
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+labels: ['type/feature', 'stage/needs-triage']
+assignees: ''
+
+---
+
+<!-- Make sure we don't have an existing Issue for that feature request (open or closed). -->
+
+### Describe your idea/feature/enhancement
+
+Provide a clear description.
```
Comment (hoffa) [inline .github/ISSUE_TEMPLATE/feature_request.md:14]: Shouldn't this be HTML comment as well like in `bug_report.md`?

**Review by hoffa — COMMENTED** [review]
Minor polishing nits.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 5 above)
Comment (torresxb1) [reply .github/ISSUE_TEMPLATE/feature_request.md:14]: Probably, but in this PR I mainly want to keep consistency across repos (as these issue templates are 99% copy-pasted across repos), so don't want to only have it differently here.  We could have follow-up PRs to update it in all the repos if necessary.

**Review by hoffa — APPROVED** [review]

**Review by qingchm — APPROVED** [review]

**Pull request merged**